nikola/shortcodes.py

```diff
@@ -43,141 +259,49 @@ def apply_shortcodes(data, registry, site=None):
-        else:  # Tag with closer
-            args[1]['data'] = data[end + 3:close - 1]
-            end = close + len(close_tag) + 1
-        pos = end
-        yield [name, args, start, end]
-
-
-def parse_sc(data):
-    """Parse shortcode arguments into a tuple."""
-    elements = data.split(' ', 1)
-    name = elements[0]
-    if len(elements) == 1:
-        # No arguments
-        return name, ([], {})
-    args = []
-    kwargs = {}
-
-    # "Simple" argument parser.
-    # flag can be one of:
-    # 0 name
-    # 1 value                               +value
-    # 2 name inside quotes                  +quotes
-    # 3 value inside quotes
-    # 4 [unsupported]                       +backslash
-    # 5 value inside backslash
-    # 4 [unsupported]
-    # 7 value inside quotes and backslash
-    flag = 0
-    cname = ''
-    cvalue = ''
-    qc = ''
-    for char in elements[1]:
-        if flag & 4 and flag & 1:
-            # Backslash in value: escape next character, no matter what
-            cvalue += char
-            flag = flag & 3
-        elif flag & 4:
-            # Backslash in name: escape next character, no matter what
-            cname += char
-            flag = flag & 3
-        elif char == '=' and flag == 0:
-            # Equals sign inside unquoted name: switch to value
-            flag = 1
-        elif char == ' ' and flag == 0:
-            # Space inside unquoted name: save as positional argument
-            args.append(cname)
-            cname = cvalue = qc = ''
-        elif char == ' ' and flag == 1:
-            # Space inside unquoted value: save as keyword argument
-            kwargs[cname] = cvalue
-            flag = 0
-            cname = cvalue = qc = ''
-        elif char == ' ' and flag == 2:
-            # Space inside quoted name: save to name
-            cname += char
-        elif char == ' ' and flag == 3:
-            # Space inside quoted value: save to value
-            cvalue += char
-        elif char == '\\':
-            # Backslash: next character will be escaped
-            flag = flag | 4
-        elif char == '"' or char == "'":
-            # Quote handler
-            qc = char
-            if not flag & 2:
-                flag += 2
-            elif flag & 2 and qc == char:
-                flag -= 2
-            elif flag == 2:
-                # Unbalanced quotes, reproduce as is
-                cname += char
-            elif flag == 3:
-                # Unbalanced quotes, reproduce as is
-                cvalue += char
-        elif flag & 1:
-            # Fallback: add anything else to value
-            cvalue += char
-        else:
-            # Fallback: add anything else to name
-            cname += char
-
-    # Handle last argument
-    if cvalue:
-        kwargs[cname] = cvalue
-    else:
-        args.append(cname)
-
-    return name, (args, kwargs)
+    empty_string = data[:0]  # same string type as data; to make Python 2 happy
```

Comment:
This makes some test fail, since it always returns a unicode string. My final fix makes it return the same string type as the input was. But we could of course also fix the tests :)